client/src/redux/reducers/handleCart.js

```diff
@@ -0,0 +1,33 @@
+const cart = [];
+
+const handleCart = (state = cart, action) => {
+    const product = action.payload
+    switch (action.type) {
+        case "ADDITEM" : 
+            const exist = state.find((x)=>x.id === product.id)
```

Comment:
for now we are looking for one element. Complexity is O(n). 
Then do iteration throw this array. Complexity is O(n). 
Let's do it for O (n). 

 ```suggestion
                if (exist){
                 return state.map((x)=>
                 x.id === product.id ? {...x, qty: x.qty+1} : x)
             } else{
             const product = action.payload
                 return[
                 ...state,
                     {
                         ...product,
                         qty:1,
                     }
                 ]
             }       
     ```